app/helpers/exports_helper.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# Helper methods for the Exports controller
+module ExportsHelper
+  def each_source_metadata_for_plate(plate)
+    index = 0
+    plate.wells_in_columns.each do |well|
+      well.transfer_requests_as_target.each do |transfer_req|
+        # NB. Making assumption here that name field on asset is for a plate well
+        # and contains a plate barcode and well location e.g. DN12345678:A1
+        name_array = transfer_req.source_asset.name.split(':')
+        if name_array.length == 2
```

Comment:
I don't fully understand this check, as I'm not quite sure why we're expecting wells wells which don't conform to this format (I guess if we see a transfer from a tube?), or, if we do come across them, why we just ignore them silently.

I know this is previous code, and I'm certainly not suggesting it be removed (as I'm assuming there *was* a reason the check was added) just wondering if you had any insight?

Reply:
I've felt the same way about it.  They are of course silently ignored.  The only reason this would not be the case is if the implementation changes for the name of the well, which I think is defined here: https://github.com/sanger/sequencescape/blob/develop/app/models/well.rb#L323

I would recommend putting a comment near where the name is defined to say that changing the format would affect this code, but as it may be in a different repo entirely that might not be a good idea.  Of course if the split didn't work and we try to reference index 1 in a single item array then we're going to generate exceptions.  I don't know if that's preferable to missing out results from the CSV.

Reply:
I'm generally one to favour blowing up loudly if we see something unexpected, or at the very least with a little bit of noise. Although I'd probably handle it specifically in this case, with a custom exception. 

But as you say, this is already coupling behaviour to another attribute in a somewhat unpredictable manner (Although I can't imagine this is the only piece of code where something is relying on this, and that goes for customer reports as well). Perhaps the time would be better spent adding the relevant associations to the API so that the information can be pulled out formally. I'll pop a story in the backlog now, as its a pretty minor thing, but its daft that you cant access a plate from its wells.

Reply:
PR with warning: https://github.com/sanger/sequencescape/pull/3343
The link is currently broken, but should resolve correctly once this is merged in.

Reply:
And for completeness the issue: https://github.com/sanger/sequencescape/issues/3342